collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java

```diff
@@ -701,7 +704,8 @@ protected void onPreExecute() {
                         nextPage.put(PARENT_ID_KEY, parentId);
                         nextPage.put(CURRENT_ID_KEY, currentDir);
                         nextPage.put(FILE_LIST_KEY, driveFileListPage);
-                        publishProgress(nextPage);
+                        filterForms(nextPage);
+                        publishProgress();
```

Comment:
Could you clarify this change? Why did you factored out `filterForms()` which contains generally the same code as `publishProgress` before and now call both?

Reply:
The filtering which was being done in `publishProgress` , i needed it for calculation of media folders and all these should run on background thread because of being computationally expensive, Whereas `publishProgress` runs on UI thread that's why.